roles/boot_iso/tasks/pxe.yml

```
@@ -0,0 +1,119 @@
+---
+
+- name: prepare PXE
+  block:
+    - name: ensure the required packgaes installed
+      package:
+        name: "{{ item }}"
+        state: present
+      with_items:
+        - syslinux
+        - tftp-server
+        - dnsmasq
+        - ipmitool
+        - httpd
+        - python3-pip
+
+    - name: Install multi python packages with version specifiers
+      pip:
+        name:
+          - pyghmi
+
+    - name: "httpd  Configuration File"
+      template:
+           src: "./httpd.conf.j2"
+           dest: "/etc/httpd/conf.d/new_conf.conf"
+    
+
+    - name: Start service httpd, if not started
+      service:
+        name: "{{ item }}"
+        state: started
+        enabled: yes
+      with_items:
+        - tftp
+        - httpd
+
+          
+    - name: mount the discovery iso
+      mount:
+        src: "/opt/http_store/data/{{ discovery_iso_name }}"
+        #src: "{{ discovery_iso_server }}/{{ discovery_iso_name }}"
+        path: /mnt
+        state: mounted
+        fstype: iso9660
+        opts: ro,noauto
+ 
+    - name: Copy mount file to tftp directory
+      copy:
+        src: "{{ item }}"
+        dest: "{{ tftp_directory }}"
+      with_items: "{{ mounted_iso_files }}"
+      vars:
+        tftp_directory: "/var/lib/tftpboot/"
+        mounted_iso_files:
+                #      - /mnt/EFI/redhat/grub.cfg
+          - /mnt/images/ignition.img
+          - /mnt/images/efiboot.img
+          - /mnt/images/pxeboot/vmlinuz
+          - /mnt/images/pxeboot/initrd.img
+   
+    - name: " grub Configuration File"
```

Maybe you can name it like
```suggestion
    - name: " Create grub configuration file"
```